modules/launchd/default.nix

```diff
@@ -0,0 +1,192 @@
+    enable = mkOption {
+      type = types.bool;
+      default = isDarwin;
+      defaultText = literalExpression "pkgs.stdenv.hostPlatform.isDarwin";
+      description = ''
+        Whether to enable Home Manager to define per-user daemons by making use
+        of launchd's LaunchAgents.
+      '';
+    };
+
+    agents = mkOption {
+      type = with types; attrsOf (submodule launchdConfig);
+      default = { };
+      description = "Define LaunchAgents.";
+    };
+  };
+
+  config = mkMerge [
+    {
+      assertions = [{
+        assertion = (cfg.enable && agentPlists != { }) -> isDarwin;
+        message = let names = lib.concatStringsSep ", " (attrNames agentPlists);
+        in "Must use Darwin for modules that require Launchd: " + names;
+      }];
+    }
+
+    (mkIf isDarwin {
+      home.extraBuilderCommands = ''
+        ln -s "${agentsDrv}" $out/LaunchAgents
+      '';
+
+      home.activation.checkLaunchAgents =
+        hm.dag.entryBefore [ "writeBoundary" ] ''
+          checkLaunchAgents() {
+            local oldDir="$(readlink -m "$oldGenPath/LaunchAgents")"
+            local newDir=${escapeShellArg agentsDrv}
+            local dstDir=${escapeShellArg dstDir}
+
+            local oldSrcPath newSrcPath dstPath agentFile agentName
+
+            find -L "$newDir" -maxdepth 1 -name '*.plist' -type f -print0 \
+                | while IFS= read -rd "" newSrcPath; do
+              agentFile="''${newSrcPath##*/}"
+              agentName="''${agentFile%.plist}"
+              dstPath="$dstDir/$agentFile"
+              oldSrcPath="$oldDir/$agentFile"
+
+              if [[ ! -e "$dstPath" ]]; then
+                continue
+              fi
+
+              if ! cmp --quiet "$oldSrcPath" "$dstPath"; then
+                errorEcho "Existing file '$dstPath' is in the way of '$newSrcPath'"
+                exit 1
+              fi
+            done
+          }
+
+          checkLaunchAgents
+        '';
+
+      # NOTE: Launch Agent configurations can't be symlinked from the Nix store
+      # because it needs to be owned by the user running it.
+      home.activation.setupLaunchAgents =
+        hm.dag.entryAfter [ "writeBoundary" ] ''
+          setupLaunchAgents() {
+            local oldDir="$(readlink -m "$oldGenPath/LaunchAgents")"
+            local newDir="$(readlink -m "$newGenPath/LaunchAgents")"
+            local dstDir=${escapeShellArg dstDir}
+            local domain="gui/$UID"
+            local err=0
+
+            local srcPath dstPath agentFile agentName
+
+            find -L "$newDir" -maxdepth 1 -name '*.plist' -type f -print0 \
+                | while IFS= read -rd "" srcPath; do
+              agentFile="''${srcPath##*/}"
+              agentName="''${agentFile%.plist}"
+              dstPath="$dstDir/$agentFile"
+
+              if cmp --quiet "$srcPath" "$dstPath"; then
+                continue
+              fi
+              if [[ -f "$dstPath" ]]; then
+                while true; do
```

Comment:
Would it be possible to limit this to just a few loops and then, e.g., `return` with an error message? It makes me a bit nervous to have a potentially infinite loop in an activation script, however unlikely it is to happen 🙂

Reply:
I set the limit to 10 loops.